src/dbnode/network/server/tchannelthrift/node/service.go

```diff
@@ -795,24 +795,48 @@ func (s *service) FetchBatchRaw(tctx thrift.Context, req *rpc.FetchBatchRawReque
 		nonRetryableErrors int
 	)
 
+	// Call ReadEncoded on each of the IDs in a loop before converting them into segments
+	// to kick off as many async fetches as possible since the "future" is not forced to
+	// resolve until the block readers are converted to a segment.
+	idBRs := make([][][]xio.BlockReader, len(req.Ids))
 	for i := range req.Ids {
 		rawResult := rpc.NewFetchRawResult_()
 		result.Elements = append(result.Elements, rawResult)
 
 		tsID := s.newID(ctx, req.Ids[i])
-		segments, rpcErr := s.readEncoded(ctx, db, nsID, tsID, start, end)
-		if rpcErr != nil {
-			rawResult.Err = rpcErr
+		encoded, err := db.ReadEncoded(ctx, nsID, tsID, start, end)
+		if err != nil {
+			rawResult.Err = convert.ToRPCError(err)
```

Comment:
`rawResult` isn't saved anywhere yeah?

The expectation is that all results come back in the order they were asked to be fetched. This Err needs a `rawResult.Err` to be set and also included in the response if an error happens here.  

Kind of surprised that tests are passing, maybe I need to go add some integration tests for this.

Here you can see I end up keep an err in this first pass and then encode the err later if the read failed:
https://github.com/m3db/m3/pull/1839/files#diff-e1587bc440ef470f59c86ac77d85a101